src/Widgets/SettingsPopover.vala

```diff
@@ -211,4 +211,18 @@ public sealed class Terminal.SettingsPopover : Gtk.Popover {
         label.set_string ("%.0f%%".printf (font_scale.get_double () * 100));
         return true;
     }
+
+    // Valid css selectors must not start with a number or contain delimiters besides `-`
+    private static string theme_to_css_class (string theme) {
+        var theme_palette = Themes.get_rgba_palette (theme);
+        var background = theme_palette[Themes.PALETTE_SIZE - 3];
+        var foreground = theme_palette[Themes.PALETTE_SIZE - 2];
+
+        var css_class = "theme-%.0lf%.0lf%.0lf%.0lf-%.0lf%.0lf%.0lf%.0lf".printf (
```

Comment:
Do we need so complicated css class?  See what I did in the Gtk4 branch - one css class for each theme name.  Colors will be overwritten if they change?

Reply:
Oh yeah theme name probably works fine. I definitely overthought it 😅